scripts/enumerable.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return return_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return return_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    select = []
+    my_each { |item| select << item if yield item }
+    select
+  end
+
+  def my_all?(class_passed = nil)
+    all = true
+    if block_given?
+      my_each { |item| all = false unless yield item }
+    elsif class_passed
+      my_each { |item| all = false unless item.is_a? class_passed }
+    else
+      my_each { |item| all = false unless item }
+    end
+    all
+  end
+
+  def my_any?(class_passed = nil)
+    any = false
+    if block_given?
+      my_each { |item| any = true if yield item }
+    elsif class_passed
+      my_each { |item| any = true if item.is_a? class_passed }
+    else
+      my_each { |item| any = true if item }
+    end
+    any
+  end
+
+  def my_none?(class_passed = nil)
+    none = true
+    if block_given?
+      my_each { |item| none = false if yield item }
+    elsif class_passed
+      my_each { |item| none = false if item.is_a? class_passed }
+    else
+      my_each { |item| none = false if item }
+    end
+    none
+  end
```

Comment:
Please check the review comment on `my_all?`. Also, you can find more details about [`none?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F)